### ISSUE TITLE
[Bug]: Changing axis direction breaks carousel.

### ISSUE THREAD

**Issue by smc13** [davidjerleke/embla-carousel#744]
### Which variants of Embla Carousel are you using?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [X] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Steps to reproduce

The bug occurs when I re-initialise the carousel with an updated axis. The carousel resets to the first slide and I can no long drag between slides.

[Code Sandbox](https://codesandbox.io/p/sandbox/embla-carousel-axis-switch-9ls23d)

### Expected Behavior

Ideally I'd like if it worked the same way as changing the direction which appears to keep its current position.

### Additional Context

I have also tried...

### Version

v8.0.0-rc22

### CodeSandbox

https://codesandbox.io/p/sandbox/embla-carousel-axis-switch-9ls23d

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for bug reports

**Comment by davidjerleke** [comment]
@smc13 thanks for your bug report. I will investigate if this is a bug or not when possible.

Best,
David

**Comment by davidjerleke** [comment]
> I cant view that sandbox (404) 

@smc13 sorry about that. I had to make the sandbox `public`. Try now!

**Comment by smc13** [comment]
I've found in Vue I can update the [watchers in the plugin](https://github.com/davidjerleke/embla-carousel/blob/b04f0afaf28fc8441e287db07922ab60db618c0c/packages/embla-carousel-vue/src/components/emblaCarouselVue.ts#L51) to add `{ flush: 'post' }` which causes the watchers to fire after any DOM updates in the parent component.

Not sure if it'll cause any timing issues though (eg: if slides and options are updated at the time time, possibly causing the watchers `reInit` and emblas native slide watcher to trigger at the same time)

**Comment by ancashoria** [comment]
I'm trying to achieve the same thing in React. Shouldn't embla handle the reactivity by default? 

```
  const axis = isMobile ? "x" : "y";
const [emblaMainRef, emblaMainApi] = useEmblaCarousel({
    axis,
  });
```

My `isMobile` hook returns the correct value asyncronously and `useEmblaCarousel` doesn't update when the axis changes

**Comment by davidjerleke** [comment]
@ancashoria did you pay attention to [this](https://github.com/davidjerleke/embla-carousel/issues/744#issuecomment-1951937131)?

**Comment by ancashoria** [comment]
I have and even tried to delay the reinit and when axis changes:
```
useEffect(() => {
    setTimeout(() => {
      emblaMainApi?.reInit({ axis });
      emblaThumbsApi?.reInit({ axis });
    }, 1000);
  }, [axis]);
```

**Comment by davidjerleke** [comment]
@ancashoria yes, but how about your CSS? Did you change it to align the slides vertically when the axis is `y`?

**Comment by ancashoria** [comment]
Thanks, it must have been something with my home made `isMobile` hook.
I have migrated to [breakpoints](https://www.embla-carousel.com/api/options/#breakpoints) and everything works smooth.

Thank you so much and congrats on the 8.0 release
